docs/proposals/user-namespace.md

```diff
@@ -0,0 +1,143 @@
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Support Host User Namespaces
+
+## Motivation
+
+Docker 1.10 supports [user namespace remapping](https://docs.docker.com/v1.10/engine/reference/commandline/daemon/#daemon-user-namespace-options)
+in the docker daemon.  This allows you to run as the root user in a
+container while running as a non-root user in the host.
+
+Additionally, the `HostConfig` for a container now contains a parameter
+that lets you escape the private user namespace and indicate that you
+would like to [use the host's user namespace](https://github.com/docker/engine-api/blob/master/types/container/host_config.go#L310).
+
+If the Docker daemon is launched with the remap option in a Kubernetes
+cluster today and a pod author would like to run a container that uses
+the host user namespace there is not currently a way to express the
+configuration.
+
+## Assumptions
+
+1.  There is currently a bug in Docker that does not allow you to use
+`--net=host`, `--pid=host`, or `--ipc=host` when remapping is enabled.
+This is fixed in https://github.com/docker/docker/pull/25771 such that
+if you are enabling user namespace remapping you may use `--net=host`, `--pid=host`,
+and `--ipc=host` flags if `--userns=host`.
+
+## When is host user namespace required
+
+Kubernetes cluster administrators wish to have a higher level of
+default cluster security on Docker 1.10+ by enabling user namespace
+remapping, but want to be able to continue to run cluster level actions
+for monitoring.  Assuming that the docker daemon is started with
+`--userns-remap=default`,
+here is when the host user namespace should be set:
+
+1.  When setting any other namespace flag to `host`
+    * Not strictly required for `net=host` if not binding to low ports
+     but if the system provides defaulting it covers the 80% case
+2.  When using capabilities that aren't tied to any namespace
+    * MKNOD
+    * SYS_MODULE
+    * SYS_TIME
+    * others?  (@mrunalp is looking at source to determine this)
+3.  When explicitly granted permissions to use a host volume and
```

Comment:
> Are you implying that using host volumes will automatically turn off userns

No.  This case is covering any niche cases where you have permissions to use a PSP that allows you to escape user ns and also use host volumes.  This would be an explicit request from the user to enter that state though, not something we could reasonably default.  Items 1 and 2 could be defaulted.